File: TD/TD6/Code/Python/4.py
#!/usr/bin/python
# Retourner l'indice de la plus petite valeur contenue dans le tableau


def recherche_min(tab):
    val_min = tab[0]

    for i in range(len(tab)):
        if tab[i] < val_min:
            val_min = tab[i]

    return tab.index(val_min)


tab = [5, 18.5, 13.2, 8.75, 2, 15, 13.5, 6, 17]

# val_min = tab.index(min(tab))  # I ❤ Python
print(recherche_min(tab))

File: TD/TD6/Code/4.cpp
/**
 * Retourner l'indice de la plus petite valeur contenue dans le tableau
 */
#include <iostream>

/* Procédure */

float rechercheMin(float tab[], int n)
{
  int index;
  float min;
  index = 0;
  min = tab[0];

  for (int i = 0; i < n; ++i)
    if (tab[i] < min) {
      min = tab[i];
      index = i;
    }

  return index;
}

/* Utilisation */

int main()
{
  float tab[9] = {5, 18.5, 13.2, 8.75, 2, 15, 13.5, 6, 17};

  std::cout << rechercheMin(tab, 9) << std::endl;

  return 0;
}
